### PR TITLE
Canonicalize file-backed KiCad footprint IDs

### PR DESCRIPTION
**Pull request by akhilles**
Before, if a footprint came from a file like: `.../Capacitor_SMD.pretty/C_0402_1005Metric.kicad_mod`

we were emitting: `C_0402_1005Metric:C_0402_1005Metric`

Now we emit: `Capacitor_SMD:C_0402_1005Metric`

and similarly:
- plain local dir `.../TLV9001IDBVR/SOT95P280X145-5N.kicad_mod` -> `TLV9001IDBVR:SOT95P280X145-5N`
- versioned package-root footprint `.../BSS138-7-F/0.3.2/SOT96P240X115-3N.kicad_mod` -> `BSS138-7-F:SOT96P240X115-3N`

---

> [!NOTE]
> **Medium Risk**
> Changes how footprints are normalized into KiCad `lib:footprint` identifiers across both Rust netlist generation and the Python layout sync, which can affect footprint library lookup and fp-lib-table contents. Logic is well-covered by new unit tests but may surface edge cases in path/package-root matching.
> 
> **Overview**
> **Canonicalizes file-backed KiCad footprint IDs** so footprints referenced by `.kicad_mod` paths (including `.pretty` libraries and `package://` URIs) emit stable `lib:fp` identifiers instead of `stem:stem`.
> 
> This updates Rust netlist export and fp-lib-table generation to use a new `format_footprint_with_package_roots` path/package-root aware formatter, mirrors the same behavior in `update_layout_file.py`, and adds focused tests covering `.pretty` paths, plain directories, and package-root/versioned URI cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ca13a4633a188c921bab2e8556fb34441638a1cb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---

<a href="https://app.devin.ai/review/diodeinc/pcb/pull/702" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: crates/pcb-sch/src/kicad_netlist.rs [hunk 1]
```diff
@@ -410,26 +411,88 @@ fn collect_pins_for_component(
...
+    {
+        let repo_name = repo.rsplit('/').next().unwrap_or(&repo);
+        return Some(repo_name.to_string());
+    }
+
+    parent
+        .file_stem()
+        .or_else(|| parent.file_name())
+        .map(|name| name.to_string_lossy().to_string())
+        .filter(|name| !name.is_empty())
```
[the remark this answers is not in the document]
Comment (akhilles) [reply crates/pcb-sch/src/kicad_netlist.rs:473]: no, this is fine.
